src/nodes/gpgpu/SubgroupFunctionNode.js

```diff
@@ -0,0 +1,191 @@
+import TempNode from '../core/TempNode.js';
+import { addMethodChaining, nodeProxy } from '../tsl/TSLCore.js';
+
+class SubgroupFunctionNode extends TempNode {
```

Comment:
Do you mind adding JSDoc for `SubgroupFunctionNode`? 

For new modules, we should add documentation right from the beginning. Use `FunctionNode` as an orientation

Reply:
For the class, how about this:

This class represents a set of built in WGSL shader functions that synchronously execute an operation across a subgroup, or 'wave', of compute or fragment shader invocations within a workgroup. Typically, these functions will synchronously execute an operation using data from all active invocations within the subgroup, then broadcast that result to all active invocations. In other graphics APIs, subgroup functions are also referred to as wave intrinsics (DirectX/HLSL) or warp intrinsics (CUDA).